pkg/storage/image.go

```diff
@@ -405,30 +349,23 @@ func (svc *imageService) PullImage(systemContext *types.SystemContext, imageName
 	if err != nil {
 		return nil, err
 	}
-	if options == nil {
-		options = &copy.Options{}
-	}
 
-	srcRef, err := svc.prepareReference(imageName, options)
+	options := *inputOptions // A shallow copy
```

Comment:
Are we no longer worried about null pointer dereference?

Reply:
https://github.com/cri-o/cri-o/pull/2543/commits/39af0992a5d1cbaf0ab8bfee084387731aa9570a :
> Expect a non-nil copy.Options in ImageServer.PullImage
>
> Both callers pass a non-nil value already.

(I originally made the `copy.Options` a struct passed by value to make this enforced by the type system, but lint was complaining about a large parameter.  If only Go had explicit optionals instead of pointers with `nil`…)

Reply:
That said, no further clean-ups depend on this, so I’d be happy to drop this change if the increased resilience is preferred.

Reply:
oops, missed the comment. If you're not worried about it, neither am I, I just wanted to make sure our expectations were mentioned at least